File: ACM/hdu/2021.10.10/1008.cpp
/*
 * @Author: NEFU AB-IN
 * @Date: 2021-10-10 12:09:01
 * @FilePath: \ACM\Hdu\2021.10.10\1008.cpp
 * @LastEditTime: 2021-10-10 14:57:29
 */
#include <bits/stdc++.h>
using namespace std;
#define LL long long
#define MP make_pair
#define SZ(X) ((int)(X).size())
#define IOS                                                                                                            \
    ios::sync_with_stdio(false);                                                                                       \
    cin.tie(0);                                                                                                        \
    cout.tie(0);
#define DEBUG(X) cout << #X << ": " << X << endl;
typedef pair<int, int> PII;
const int INF = 0x3f3f3f3f;

signed main()
{
    IOS;
    int ans = 0;
    int a[5] = {0, 1, 2, 3, 4};
    do
    {
        for (int i = 1; i <= 4; ++i)
        {
            cout << a[i];
        }
        ans++;
    } while (next_permutation(a + 1, a + 1 + 4));

    return 0;
}

File: ACM/hdu/2021.10.10/1008.py
'''
Author: NEFU AB-IN
Date: 2021-10-10 15:05:01
FilePath: \ACM\Hdu\2021.10.10\1008.py
LastEditTime: 2021-10-10 15:21:57
'''
import itertools as it


def f(n, k):
    l = []
    for i in range(1, n + 1):
        l.append(i)
    lst = list(it.permutations(l))
    ll = []
    for i in lst:
        ll += i
    ####
    l = []
    for i in range(1, k + 1):
        l.append(i)
    lst = list(it.permutations(l))
    ans = 0
    for i in range(len(ll) - k + 1):
        if tuple(ll[i:i + k]) in lst:
            ans += 1
    print(ans, end=" ")


for n in range(1, 10):
    for k in range(1, n + 1):
        f(n, k)
    print()
